stage0/src/asm/boot.s

```diff
@@ -23,18 +23,46 @@ _start :
 
 .align 16
 .code32
+.global vc_handler
+vc_handler:
+    # For now, let's assume that the #VC interrupt is triggered when we call the CPUID instruction
+    # in the SEV detection code, below, while running under SEV-{ES, SNP}. Thus, the fact that we're
+    # in this handler altogether will suffice as evidence that we support the SEV_STATUS MSR.
+    # It's probably not worth implementing a proper interrupt handler here (that knows how to use the
+    # GHCB protocol) as it's only used in the bootstrap code.
+    pop %eax        # ignore the error code for now
+    pop %eax        # pop the return address
+    add $2, %eax    # increment it by 2 (size of the CPUID instruction)
+    push %eax       # push it back on stack for iret
+    mov $0xA, %eax  # return value for cpuid saying memory encryption is enabled
+    iret            # go back
+
 _protected_mode_start:
     # Switch to a flat 32-bit data segment, giving us access to all 4G of memory.
     mov $ds, %eax
     mov %eax, %ds
+    mov %eax, %ss
+
+    # Set up a basic stack, as we may get interrupts.
+    mov $stack_start, %esp
 
-    # Determine if we're running under SEV
+    # Determine if we're running under SEV. Keep track of which bit is the encrypted bit in %rsi.
+    mov $0, %esi              # by default, no encryption
+    mov $0x8000001F, %eax
```

Comment:
Any reference for these numbers? I'm thinking if you are OOO and somehow we get errors that seem related to these numbers, how would we (anyone other than you) double check them? :)

Reply:
You ask @conradgrobler of course :)

I've added some documentation.